x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_link.ts

```diff
@@ -4,10 +4,28 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { BASE_PATH } from '../constants';
+import {
+  BASE_PATH,
+  StaticPage,
+  DynamicPage,
+  DynamicPagePathValues,
+  pagePathGetters,
+} from '../constants';
 import { useCore } from './';
 
-export function useLink(path: string = '/') {
+const getPath = (page: StaticPage | DynamicPage, values?: DynamicPagePathValues): string => {
+  return values ? pagePathGetters[page](values) : pagePathGetters[page as StaticPage]();
+};
+
+export const useLink = () => {
   const core = useCore();
-  return core.http.basePath.prepend(`${BASE_PATH}#${path}`);
-}
+  return {
+    getPath: (page: StaticPage | DynamicPage, values?: DynamicPagePathValues) => {
```

Comment:
why not returning `getPath` directly? we can avoid to create a new function

Reply:
good catch, leftover logic from when I DRY'd `getPath` 😅